plugins/backend/ttrss/ttrssAPI.vala

```diff
@@ -212,22 +212,16 @@ public class FeedReader.ttrssAPI : GLib.Object {
 					{
 						var feed_node = response.get_object_element(i);
 						string feed_id = feed_node.get_int_member("id").to_string();
-
-						bool hasIcon = feed_node.get_boolean_member("has_icon");
-
-						if(hasIcon)
-							Utils.downloadIcon(feed_id, m_iconDir + feed_id + ".ico");
-						else
-							hasIcon = Utils.downloadFavIcon(feed_id, feed_node.get_string_member("feed_url"));
+						string? icon_url = feed_node.get_boolean_member("has_icon") ? m_iconDir+feed_id+".ico" : null;
```

Comment:
m_iconDir is the base url to the ttrss-instance (in my case `http://nas/tt-rss/`) plus the `icons_url` from the API. So together something like:
`http://nas/tt-rss/path/to/icons/`
To download the icon for the feed with ID 23 we just add `23.ico`.
The complete url for feed 23 would therefore be
`http://nas/tt-rss/path/to/icons/23.ico`